packages/slate-react/src/components/leaf.js

```diff
@@ -49,20 +55,59 @@ class Leaf extends React.Component {
     debug(message, `${this.props.node.key}-${this.props.index}`, ...args)
   }
 
+  forceRegeneration = () => {
+    this.setState(state => ({
+      regenerateNum: state.regenerateNum + 1,
+    }))
```

Comment:
how is this different than `this.forceUpdate()`?

Reply:
@CameronAckermanSEL Thank you. I just find the re-mount in leaf.js is un-necessary. The remount is necessary in text.js where the dom element of leaf can be removed.

The difference of forceUpdate and set a different key is that, by setting a different key, the react will unmount the current dom without re-using the childNodes.  It is useful in spell check when the childNodes of text.js can be removed without notifying react.

Reply:
My mistake. The renerageKey of leaf.js is useful.

See this image:
![screenshot_20180307_235202](https://user-images.githubusercontent.com/746159/37930272-45c3dc26-3110-11e8-8a8b-7787fd868559.png)

under the DOM generated after spell check.  The `try it out` and `help` are not controlled by react.  We have to regenerate key to unmount this whole leaf dom and mount a new leaf dom.